ui/app/components/two-step-button.js

```diff
@@ -22,5 +24,10 @@ export default Component.extend({
     promptForConfirmation() {
       this.set('state', 'prompt');
     },
+    confirm() {
+      RSVP.resolve(this.get('onConfirm')()).then(() => {
+        this.send('setToIdle');
+      });
+    },
```

Comment:
From the usage I've seen, `awaitingConfirmation` is always tied to the task used in `onConfirm`. Would it make sense to structure this as a wrapping ember-concurrency task instead? Sketch:

```hbs
{{two-step-button
  idleText="Scary Action"
  cancelText="Nvm"
  confirmText="Yep"
  confirmationMessage="Wait, really? Like...seriously?"
  onConfirm=(perform yesReallyDoItNow)}}
```

```js
export default Component.extend({
  confirm: task(function*() {
    yield this.onConfirm();
  }).drop(),
  awaitingConfirmation: alias('confirm.isRunning'),
});
```

Less flexible (so maybe not suitable), but a simpler API.

Reply:
Interesting! I've never operated on the result of the `perform` helper like that. I like that the `awaitingConfirmation` detail is no longer a concern of the user of a component, but I'm not sure if it's safe to assume that the component will always make use of the loading state, (e.g., clearing an editor or form).

I think I'll keep it the way it is for now, but this looks like a cool pattern.